### PR TITLE
:book: Relax our backporting guidelines and policies

### PR DESCRIPTION
**Pull request by vincepri**
This change relaxes and defines a bit more what can be backported and
how. After this change is merged, we'll allow both bug fixes and minor
(non-breaking) features to be backported if at least one or more
maintainers approves the cherry pick.

With these changes we aim to be more welcoming of changes to the current
stable branch instead of stopping the world until the next releases.

The only caveat is that backports that I'd like to highlight is that
backporting is an expensive operation. More often that not,
a cherry-picked commit requires another full review because changes are
usually not compatible.

Going forward, we won't allow PRs to target a release branch directly,
we'll always require changes to go in the main branch first, and only
after they've been merged, to backport them.

Signed-off-by: Vince Prignano <vincepri@vmware.com>

/assign @CecileRobertMichon @ncdc @detiber

### CODE REVIEW

**Comment by CecileRobertMichon** [comment]
+1 from me

/hold to discuss at office hours

File: CONTRIBUTING.md
Comment (detiber) [inline CONTRIBUTING.md:155]: Since this is getting updated anyway, it might be good to go ahead and update this to the current active release branch rather than continuing to reference `release-0.2` here.
```suggestion
  Let's assume that the most recent release branch is `release-0.3`
```

File: CONTRIBUTING.md
Comment (detiber) [inline CONTRIBUTING.md:157]:
```suggestion
  A pull request that has been merged in the main branch can be backported to the `release-0.3`
```

**Review by detiber — COMMENTED** [review]
minor nit, otherwise lgtm

**Comment by ncdc** [comment]
LGTM pending Jason's suggestion

**Comment by fabriziopandini** [comment]
lgtm for me

**Comment by vincepri** [comment]
This should be good to go, we discussed it at the community meeting as well

/hold cancel

**Review by detiber — COMMENTED** [review]
/lgtm

**Review by CecileRobertMichon — APPROVED** [review]
/lgtm
/approve

**Pull request merged**
